File: core/src/testFixtures/resources/fixtures/repos/gradle-small/src/main/kotlin/pay/Router.kt
package pay

public data class Request(val path: String, val userId: Int? = null)

public data class Response(val status: Int, val body: String)

public val CURRENCIES: List<String> = listOf("EUR", "USD", "GBP")

public fun normalizeCurrency(raw: String): String {
    val code = raw.trim().uppercase()
    require(code in CURRENCIES) { "unknown currency: $raw" }
    return code.lowercase()
}

public class Router(private val routes: Map<String, (Request) -> Response>) {
    public fun route(req: Request): Response =
        routes[req.path]?.invoke(req) ?: Response(404, "no route for ${req.path}")

    public fun paths(): List<String> = routes.keys.sorted()
}

File: core/src/testFixtures/resources/fixtures/repos/ts-small/src/router.ts
export type Request = { path: string; userId?: number };

export type Response = { status: number; body: string };

export type Handler = (req: Request) => Response;

export const CURRENCIES = ["EUR", "USD", "GBP"] as const;

export function normalizeCurrency(raw: string): string {
  const code = raw.trim().toUpperCase();
  if (!(CURRENCIES as readonly string[]).includes(code)) {
    throw new Error(`unknown currency: ${raw}`);
  }
  return code.toLowerCase();
}

export class Router {
  private readonly routes: Map<string, Handler>;

  constructor(routes: Record<string, Handler>) {
    this.routes = new Map(Object.entries(routes));
  }

  route(req: Request): Response {
    const handler = this.routes.get(req.path);
    if (handler === undefined) {
      return { status: 404, body: `no route for ${req.path}` };
    }
    return handler(req);
  }

  paths(): string[] {
    return [...this.routes.keys()].sort();
  }
}
